diff --git a/Source/NSException.m b/Source/NSException.m
index ea4bfa411..0f3e3db73 100644
--- a/Source/NSException.m
+++ b/Source/NSException.m
@@ -1483,6 +1483,7 @@ callUncaughtHandler(id value)
   NSException	*except;
 
   reason = [NSString stringWithFormat: format arguments: argList];
+  NSLog(@"######################### name = %@, reason = %@", name, reason);
   except = [self exceptionWithName: name reason: reason userInfo: nil];
   [except raise];
   while (1);    // does not return
