diff --git a/WinNSMenu.m b/WinNSMenu.m
index 7668f2d..7b1d52f 100644
--- a/WinNSMenu.m
+++ b/WinNSMenu.m
@@ -119,15 +119,17 @@ static NSLock *menuLock = nil;
 @end
 
 @interface NSWindow (WinMenuPrivate)
-- (GSWindowDecorationView *) windowView;
+// - (GSWindowDecorationView *) windowView;
 - (void) _setMenu: (NSMenu *) menu;
 @end
 
 @implementation NSWindow (WinMenuPrivate)
+/*
 - (GSWindowDecorationView *) windowView
 {
   return _wv;
 }
+*/
 
 - (void) _setMenu: (NSMenu *) menu
 {
diff --git a/WinNSOpenPanel.m b/WinNSOpenPanel.m
index 8456774..7f19281 100644
--- a/WinNSOpenPanel.m
+++ b/WinNSOpenPanel.m
@@ -27,13 +27,13 @@
 #import <Foundation/Foundation.h>
 #import <AppKit/AppKit.h>
 
-#ifdef __MINGW64__
+// #ifdef __MINGW64__
 #define interface struct
-#endif
+// #endif
 #import <shlobj.h>
-#ifdef __MINGW64__
+// #ifdef __MINGW64__
 #undef interface
-#endif
+// #endif
 
 #import <commdlg.h>
 #import <windows.h>
@@ -338,7 +338,8 @@ unsigned long long unilen(unichar *chars)
  */
 - (NSString*) filename
 {
-  ASSIGN(_fullFileName, filename);
+  [super setFullFileName: filename];
+  // ASSIGN(_fullFileName, filename);
   return [super filename];
 }
 
@@ -528,7 +529,8 @@ unsigned long long unilen(unichar *chars)
  */
 - (NSString*) filename
 {
-  ASSIGN(_fullFileName, filename);
+  [super setFullFileName: filename];
+  // ASSIGN(_fullFileName, filename);
   return [super filename];
 }
 
diff --git a/WinUXTheme.m b/WinUXTheme.m
index aa3092a..99591e2 100644
--- a/WinUXTheme.m
+++ b/WinUXTheme.m
@@ -42,7 +42,7 @@ static inline RECT GSViewRectToWin(NSWindow *win, NSRect r)
   r = (NSRect)[ctm rectInMatrixSpace: r];
   return GSWindowRectToMS(win, r);
 }
-
+/*
 @interface NSMenu (WinUXThemePrivate)
 - (NSWindow *) window;
 @end
@@ -53,7 +53,7 @@ static inline RECT GSViewRectToWin(NSWindow *win, NSRect r)
   return _aWindow;
 }
 @end
-
+*/ 
 @implementation WinUXTheme
 
 + (void) initialize
@@ -197,12 +197,12 @@ static inline RECT GSViewRectToWin(NSWindow *win, NSRect r)
   wchar_t* classNameChars;
   
   hWnd = (HWND)[[[NSView focusView] window] windowNumber];
-  classNameChars = objc_calloc([className length]+1, sizeof(wchar_t));
+  classNameChars = calloc([className length]+1, sizeof(wchar_t));
   [className getCharacters:classNameChars];
   
   hTheme = OpenThemeData(hWnd, classNameChars);
   
-  objc_free(classNameChars);
+  free(classNameChars);
   return hTheme;
 }
 
