diff --git a/Source/GSTrackingRect.m b/Source/GSTrackingRect.m
index dc805af86..1971ca87a 100644
--- a/Source/GSTrackingRect.m
+++ b/Source/GSTrackingRect.m
@@ -142,3 +142,12 @@
 }
 
 @end
+
+@implementation GSTrackingRect (_GSPrivate_)
+
+- (NSRect) rectangle
+{
+  return rectangle;
+}
+
+@end
diff --git a/Source/NSView.m b/Source/NSView.m
index 21b3c3b0c..af3f41bc4 100644
--- a/Source/NSView.m
+++ b/Source/NSView.m
@@ -5204,3 +5204,17 @@ cmpFrame(id view1, id view2, void *context)
 }
 
 @end
+
+@implementation NSView (_GSPrivate_)
+
+- (struct _rFlagsType) rFlags
+{
+  return _rFlags;
+}
+
+- (NSArray *) cursorRects
+{
+  return _cursor_rects;
+}
+
+@end
