diff --git a/Source/GSFontInfo.m b/Source/GSFontInfo.m
index cb667faad..bce97aa3f 100644
--- a/Source/GSFontInfo.m
+++ b/Source/GSFontInfo.m
@@ -868,3 +868,146 @@ static GSFontEnumerator *sharedEnumerator = nil;
 }
 
 @end
+
+@implementation GSFontEnumerator (_GSPrivate_)
+
+- (NSMutableDictionary *) allFontFamilies
+{
+  return allFontFamilies;
+}
+
+- (NSArray *) allFontNames
+{
+  return allFontNames;
+}
+
+- (void) setAllFontFamilies: (NSMutableDictionary *)d
+{
+  allFontFamilies = d;
+}
+
+- (void) setAllFontNames: (NSArray *)a
+{
+  allFontNames = a;
+}
+
+@end
+
+@implementation GSFontInfo (_GSPrivate_)
+
+- (void) setFontName: (NSString *)n
+{
+  ASSIGN(fontName, n);
+}
+
+- (NSString *) fontName
+{
+  return fontName;
+}
+
+- (void) setMatrix: (const CGFloat *)m
+{
+   memcpy(matrix, m, sizeof(matrix));
+}
+
+- (const CGFloat *)matrix
+{
+  return matrix;
+}
+
+- (void) setFamilyName: (NSString *)n
+{
+  ASSIGN(familyName, n);
+}
+
+- (NSString *) familyName
+{
+  return familyName;
+}
+
+- (void) setCoveredCharacterSet: (NSCharacterSet *)c
+{
+  ASSIGN(coveredCharacterSet, c);
+}
+
+- (NSCharacterSet *) coveredCharacterSet
+{
+  return coveredCharacterSet;
+}
+
+- (void) setNumberOfGlyphs: (unsigned)n
+{
+  numberOfGlyphs = n;
+}
+
+- (unsigned) numberOfGlyphs
+{
+  return numberOfGlyphs;
+}
+
+- (void) setAscender: (CGFloat)a
+{
+  ascender = a;
+}
+
+- (CGFloat) ascender
+{
+  return ascender;
+}
+
+- (void) setDescender: (unsigned)d
+{
+  descender = d;
+}
+
+- (CGFloat) descender
+{
+  return descender;
+}
+
+- (void) setFixedPitch: (BOOL)f
+{
+  isFixedPitch = f;
+}
+
+- (BOOL) isFixedPitch
+{
+  return isFixedPitch;
+}
+
+- (void) setBaseFont: (BOOL)f
+{
+  isBaseFont = f;
+}
+
+- (BOOL) isBaseFont
+{
+  return isBaseFont;
+}
+
+- (void) setXHeight: (CGFloat)h
+{
+  xHeight = h;
+}
+
+- (CGFloat) xHeight
+{
+  return xHeight;
+}
+
+- (void) setMaximumAdvancment: (NSSize)s
+{
+  maximumAdvancement = s;
+}
+
+- (NSSize) maximumAdvancement
+{
+  return maximumAdvancement;
+}
+
+- (void) setFontBBox: (NSRect)r
+{
+  fontBBox = r;
+}
+
+@end
diff --git a/Source/GSTrackingRect.m b/Source/GSTrackingRect.m
index dc805af86..1971ca87a 100644
--- a/Source/GSTrackingRect.m
+++ b/Source/GSTrackingRect.m
@@ -142,3 +142,12 @@
 }
 
 @end
+
+@implementation GSTrackingRect (_GSPrivate_)
+
+- (NSRect) rectangle
+{
+  return rectangle;
+}
+
+@end
diff --git a/Source/NSView.m b/Source/NSView.m
index 21b3c3b0c..af3f41bc4 100644
--- a/Source/NSView.m
+++ b/Source/NSView.m
@@ -5204,3 +5204,17 @@ cmpFrame(id view1, id view2, void *context)
 }
 
 @end
+
+@implementation NSView (_GSPrivate_)
+
+- (struct _rFlagsType) rFlags
+{
+  return _rFlags;
+}
+
+- (NSArray *) cursorRects
+{
+  return _cursor_rects;
+}
+
+@end
