diff --git a/os/generic_win_base.h b/os/generic_win_base.h
index 5f1ea74..bcdf092 100644
--- a/os/generic_win_base.h
+++ b/os/generic_win_base.h
@@ -18,16 +18,20 @@
 // Unices provide `roundup` via sys/param.h
 #define roundup(x, y) ((((x) + ((y) - 1)) / (y)) * (y))
 // Unices provide `MAX` via sys/param.h
+#ifndef MAX
 #define MAX(a,b) (((a)>(b))?(a):(b))
+#endif
 // Unices provide `MIN` via sys/param.h
+#ifndef MIN
 #define MIN(a,b) (((a)<(b))?(a):(b))
+#endif
 // Unices provide `howmany` via sys/param.h
 #define howmany(x, y)  (((x) + ((y) - 1)) / (y))
 
 #ifndef HAVE_MODE_T
 typedef int mode_t;
 #endif
-typedef void pthread_attr_t;
+typedef struct pthread_attr_t_ * pthread_attr_t;
 
 #ifndef API_AVAILABLE
 #define API_AVAILABLE(...)
