diff --git a/src/shims/tsd.h b/src/shims/tsd.h
index 446c4d7..f8b5ec9 100644
--- a/src/shims/tsd.h
+++ b/src/shims/tsd.h
@@ -158,7 +158,7 @@ struct dispatch_tsd {
 extern _Thread_local struct dispatch_tsd __dispatch_tsd;
 
 extern void libdispatch_tsd_init(void);
-extern void _libdispatch_tsd_cleanup(void *ctx);
+extern void DISPATCH_TSD_DTOR_CC _libdispatch_tsd_cleanup(void *ctx);
 
 DISPATCH_ALWAYS_INLINE
 static inline struct dispatch_tsd *
