https://github.com/apple/swift-corelibs-libdispatch/pull/803
---
diff --git a/cmake/modules/DispatchWindowsSupport.cmake b/cmake/modules/DispatchWindowsSupport.cmake
index 87675a7..750b3be 100644
--- a/cmake/modules/DispatchWindowsSupport.cmake
+++ b/cmake/modules/DispatchWindowsSupport.cmake
@@ -6,7 +6,7 @@ function(dispatch_windows_arch_spelling arch var)
     set(${var} x64 PARENT_SCOPE)
   elseif(${arch} STREQUAL armv7)
     set(${var} arm PARENT_SCOPE)
-  elseif(${arch} STREQUAL aarch64)
+  elseif(${arch} STREQUAL aarch64 OR ${arch} STREQUAL ARM64)
     set(${var} arm64 PARENT_SCOPE)
   else()
     message(FATAL_ERROR "do not know MSVC spelling for ARCH: `${arch}`")
