diff --git a/CMakeLists.txt b/CMakeLists.txt
index 327c2a42..26ccfabf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -362,7 +362,7 @@ endif()
 
 if(LIBXML2_WITH_ICU)
     target_link_libraries(LibXml2 PRIVATE ICU::uc)
-    set(ICU_LDFLAGS "-licuuc")
+    set(ICU_LDFLAGS "-licu")
     list(APPEND XML_PRIVATE_LIBS "${ICU_LDFLAGS}")
     pkg_check_modules(ICU_PC IMPORTED_TARGET icu-uc)
     if(ICU_PC_FOUND)
@@ -416,7 +416,6 @@ set_target_properties(
     IMPORT_PREFIX lib
     OUTPUT_NAME xml2
     POSITION_INDEPENDENT_CODE ON
-    PREFIX lib
     VERSION ${PROJECT_VERSION}
     SOVERSION ${LIBXML_SOVERSION}
 )
diff --git a/encoding.c b/encoding.c
index a90694f5..a0b9a9c1 100644
--- a/encoding.c
+++ b/encoding.c
@@ -48,7 +48,7 @@
 #include "private/memory.h"
 
 #ifdef LIBXML_ICU_ENABLED
-#include <unicode/ucnv.h>
+#include <icu.h>
 #endif
 
 #define XML_HANDLER_STATIC (1 << 0)
diff --git a/libxml2-config.cmake.cmake.in b/libxml2-config.cmake.cmake.in
index de084bdf..39d30622 100644
--- a/libxml2-config.cmake.cmake.in
+++ b/libxml2-config.cmake.cmake.in
@@ -124,7 +124,7 @@ if(NOT LIBXML2_SHARED)
     endif()
 
     if(LIBXML2_WITH_ICU)
-        find_dependency(ICU COMPONENTS data i18n uc)
+        find_dependency(ICU COMPONENTS i18n uc)
         list(APPEND LIBXML2_LIBRARIES    ${ICU_LIBRARIES})
         if(NOT ICU_FOUND)
             set(${CMAKE_FIND_PACKAGE_NAME}_FOUND FALSE)
diff --git a/libxml2-config.cmake.in b/libxml2-config.cmake.in
index 31036805..469d110b 100644
--- a/libxml2-config.cmake.in
+++ b/libxml2-config.cmake.in
@@ -76,9 +76,9 @@ if(LIBXML2_WITH_THREADS)
 endif()
 
 if(LIBXML2_WITH_ICU)
-    find_dependency(ICU COMPONENTS data i18n uc)
+    find_dependency(ICU COMPONENTS i18n uc)
     list(APPEND LIBXML2_LIBRARIES    ${ICU_LIBRARIES})
-    list(APPEND LIBXML2_INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:ICU::data>;\$<LINK_ONLY:ICU::i18n>;\$<LINK_ONLY:ICU::uc>")
+    list(APPEND LIBXML2_INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:ICU::i18n>;\$<LINK_ONLY:ICU::uc>")
     if(NOT ICU_FOUND)
         set(${CMAKE_FIND_PACKAGE_NAME}_FOUND FALSE)
         set(${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE "ICU dependency was not found")
