diff --git a/include/libxml/encoding.h b/include/libxml/encoding.h
index c875af6f..1ddc9e30 100644
--- a/include/libxml/encoding.h
+++ b/include/libxml/encoding.h
@@ -28,7 +28,7 @@
 #include <iconv.h>
 #endif
 #ifdef LIBXML_ICU_ENABLED
-#include <unicode/ucnv.h>
+#include <icu.h>
 #endif
 #ifdef __cplusplus
 extern "C" {
diff --git a/win32/Makefile.msvc b/win32/Makefile.msvc
index f6a0182d..0dce87c0 100644
--- a/win32/Makefile.msvc
+++ b/win32/Makefile.msvc
@@ -73,7 +73,7 @@ LIBS = $(LIBS) iconv.lib
 !if "$(STATIC)" == "1"
 LIBS = $(LIBS) advapi32.lib sicuuc.lib sicuin.lib sicudt.lib
 !else
-LIBS = $(LIBS) icuuc.lib icuin.lib icudt.lib
+LIBS = $(LIBS) icu.lib
 !endif
 !endif
 !if "$(WITH_ZLIB)" == "1"
