diff --git a/encoding.c b/encoding.c
index 5465359f..7fb09e4c 100644
--- a/encoding.c
+++ b/encoding.c
@@ -44,7 +44,7 @@
 #include "enc.h"
 
 #ifdef LIBXML_ICU_ENABLED
-#include <unicode/ucnv.h>
+#include <icu.h>
 /* Size of pivot buffer, same as icu/source/common/ucnv.cpp CHUNK_SIZE */
 #define ICU_PIVOT_BUF_SIZE 1024
 typedef struct _uconv_t uconv_t;
diff --git a/win32/Makefile.msvc b/win32/Makefile.msvc
index f6a0182d..0dce87c0 100644
--- a/win32/Makefile.msvc
+++ b/win32/Makefile.msvc
@@ -73,7 +73,7 @@ LIBS = $(LIBS) iconv.lib
 !if "$(STATIC)" == "1"
 LIBS = $(LIBS) advapi32.lib sicuuc.lib sicuin.lib sicudt.lib
 !else
-LIBS = $(LIBS) icuuc.lib icuin.lib icudt.lib
+LIBS = $(LIBS) icu.lib
 !endif
 !endif
 !if "$(WITH_ZLIB)" == "1"
