diff --git a/Source/GNUmakefile b/Source/GNUmakefile
index 490d86690..4430c553b 100644
--- a/Source/GNUmakefile
+++ b/Source/GNUmakefile
@@ -82,12 +82,12 @@ libgnustep-baseadd_SUBPROJECTS += Additions
 
 ifeq ($(GNUSTEP_TARGET_OS), mingw32)
   libgnustep-base_SUBPROJECTS += win32
+else ifeq ($(GNUSTEP_TARGET_OS), mingw64)
+  libgnustep-base_SUBPROJECTS += win32
+else ifeq ($(GNUSTEP_TARGET_OS), windows)
+  libgnustep-base_SUBPROJECTS += win32
 else
-  ifeq ($(GNUSTEP_TARGET_OS), mingw64)
-    libgnustep-base_SUBPROJECTS += win32
-  else
-    libgnustep-base_SUBPROJECTS += unix
-  endif
+  libgnustep-base_SUBPROJECTS += unix
 endif
 
 DEFS+= -DGNUSTEP_TARGET_DIR=\"$(GNUSTEP_TARGET_DIR)\" \
