diff --git a/Headers/Foundation/NSObjCRuntime.h b/Headers/Foundation/NSObjCRuntime.h
index f3b25c177..283fc8c8e 100644
--- a/Headers/Foundation/NSObjCRuntime.h
+++ b/Headers/Foundation/NSObjCRuntime.h
@@ -297,4 +297,9 @@ DEFINE_BLOCK_TYPE(NSComparator, NSComparisonResult, id, id);
 }
 #endif
 
+/* Undefine "interface" defined in Visual Studio MSVC headers. */
+#if defined(_MSC_VER) && defined(interface)
+#undef interface
+#endif
+
 #endif /* __NSObjCRuntime_h_GNUSTEP_BASE_INCLUDE */
