diff --git a/Source/NSBundle.m b/Source/NSBundle.m
index 37da561b7..d4985f771 100644
--- a/Source/NSBundle.m
+++ b/Source/NSBundle.m
@@ -710,6 +710,8 @@ _find_main_bundle_for_tool(NSString *toolName)
           return bundle;
         }
 
+      NSWarnLog(@"\t *** frameworkClass = %@, _byClass = %@", _frameworkClass, _byClass);
+      
       name = [NSString stringWithUTF8String: &frameworkClassName[12]];
       /* Important - gnustep-make mangles framework names to encode
        * them as ObjC class names.  Here we need to demangle them.  We
@@ -726,6 +728,8 @@ _find_main_bundle_for_tool(NSString *toolName)
        */
       bundlePath = GSPrivateSymbolPath (frameworkClass, NULL);
 
+      NSWarnLog(@"\t *** name = %@, bundlePath = %@", name, bundlePath);
+
       if ([bundlePath isEqualToString: GSPrivateExecutablePath()])
 	{
 	  /* Oops ... the NSFramework_xxx class is linked in the main
@@ -783,7 +787,7 @@ _find_main_bundle_for_tool(NSString *toolName)
 	    {
 	      bundlePath = [bundlePath stringByDeletingLastPathComponent];
 	    }
-#if defined(_WIN32)
+#if defined(_WIN32) || defined(_MSC_VER)
 	  /* On windows, the library (dll) is in the Tools area rather than
 	   * in the framework, so we can adjust the path here.
 	   */
@@ -818,7 +822,7 @@ _find_main_bundle_for_tool(NSString *toolName)
 		  if (bundlePath)
 		    bundle = [[self alloc] initWithPath: bundlePath];
 		}
-#if !defined(_WIN32)
+#if !defined(_WIN32) && !defined(_MSC_VER)
 	    }
 #endif
 
