diff --git a/configure b/configure
index 662f1b6b4..5f91a2fad 100755
--- a/configure
+++ b/configure
@@ -7805,27 +7805,35 @@ fi
 #--------------------------------------------------------------------
 # Check for thread synchronisation support in runtime
 #--------------------------------------------------------------------
-for ac_func in objc_sync_enter
-do :
-  ac_fn_c_check_func "$LINENO" "objc_sync_enter" "ac_cv_func_objc_sync_enter"
-if test "x$ac_cv_func_objc_sync_enter" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
+if test "$host_os" = windows -a "$OBJC_RUNTIME_LIB" = "ng"; then
+  # On Windows MSVC, AC_CHECK_FUNCS() will throw linker errors ("relocation
+  # against symbol in discarded section") as the file won't contain any actual
+  # ObjC code, so we just assume the function is there.
+  { $as_echo "$as_me:${as_lineno-$LINENO}: assuming objc_sync_enter() on Windows MSVC" >&5
+$as_echo "$as_me: assuming objc_sync_enter() on Windows MSVC" >&6;}
+  OBJCSYNC=1
+else
+  for ac_func in objc_sync_enter
+  do :
+    ac_fn_c_check_func "$LINENO" "objc_sync_enter" "ac_cv_func_objc_sync_enter"
+    if test "x$ac_cv_func_objc_sync_enter" = xyes; then :
+      cat >>confdefs.h <<_ACEOF
 #define HAVE_OBJC_SYNC_ENTER 1
 _ACEOF
 
-fi
-done
+    fi
+  done
 
-if test $ac_cv_func_objc_sync_enter = yes ; then
-  OBJCSYNC=1
-else
-  OBJCSYNC=0
-  if test "$OBJC_RUNTIME_LIB" = "ng"; then
-    as_fn_error $? "The objc runtime library does not appear to have synchronisation support.  Try re-configuring gnustep-make with a CPPFLAGS variable containing a -L point to specify the directory containing the correct libobjc, or using the --with-objc-lib-flag=... option." "$LINENO" 5
+  if test $ac_cv_func_objc_sync_enter = yes ; then
+    OBJCSYNC=1
+  else
+    OBJCSYNC=0
+    if test "$OBJC_RUNTIME_LIB" = "ng"; then
+      as_fn_error $? "The objc runtime library does not appear to have synchronisation support.  Try re-configuring gnustep-make with a CPPFLAGS variable containing a -L point to specify the directory containing the correct libobjc, or using the --with-objc-lib-flag=... option." "$LINENO" 5
+    fi
   fi
 fi
 
-
 # Don't revert any Objective-C flags as they are used in the next test
 
 #--------------------------------------------------------------------
@@ -7835,6 +7843,13 @@ if test "$cross_compiling" = "yes"; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: \"Cross compiling! Using predefined OBJC2RUNTIME variable\"" >&5
 $as_echo "$as_me: \"Cross compiling! Using predefined OBJC2RUNTIME variable\"" >&6;}
   OBJC2RUNTIME="$cross_objc2_runtime"
+elif test "$host_os" = windows -a "$OBJC_RUNTIME_LIB" = "ng"; then
+  # On Windows MSVC, AC_CHECK_FUNCS() will throw linker errors ("relocation
+  # against symbol in discarded section") as the file won't contain any actual
+  # ObjC code, so we just assume the function is there.
+  { $as_echo "$as_me:${as_lineno-$LINENO}: assuming objc_setProperty() on Windows MSVC" >&5
+$as_echo "$as_me: assuming objc_setProperty() on Windows MSVC" >&6;}
+  OBJC2RUNTIME=1  
 else
   for ac_func in objc_setProperty
 do :
@@ -7861,76 +7876,92 @@ fi
 #--------------------------------------------------------------------
 # Check for blocks support in runtime
 #--------------------------------------------------------------------
-for ac_func in _Block_copy
-do :
-  ac_fn_c_check_func "$LINENO" "_Block_copy" "ac_cv_func__Block_copy"
-if test "x$ac_cv_func__Block_copy" = xyes; then :
+if test "$host_os" = windows -a "$OBJC_RUNTIME_LIB" = "ng"; then
+  # On Windows MSVC, AC_CHECK_FUNCS() will throw linker errors ("relocation
+  # against symbol in discarded section") as the file won't contain any actual
+  # ObjC code, so we just assume the function is there.
+  { $as_echo "$as_me:${as_lineno-$LINENO}: assuming _Block_copy() on Windows MSVC" >&5
+$as_echo "$as_me: assuming _Block_copy() on Windows MSVC" >&6;}
+  HAVE_BLOCKS=1
+else
+  for ac_func in _Block_copy
+    do :
+      ac_fn_c_check_func "$LINENO" "_Block_copy" "ac_cv_func__Block_copy"
+      if test "x$ac_cv_func__Block_copy" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE__BLOCK_COPY 1
 _ACEOF
 
-fi
-done
+      fi
+  done
 
-if test $ac_cv_func__Block_copy = yes ; then
-  HAVE_BLOCKS=1
-else
-  HAVE_BLOCKS=0
-  if test "$OBJC_RUNTIME_LIB" = "ng"; then
-    as_fn_error $? "The objc runtime library does not appear to have blocks support.  Try re-configuring gnustep-make with a CPPFLAGS variable containing a -L point to specify the directory containing the correct libobjc, or using the --with-objc-lib-flag=... option." "$LINENO" 5
+  if test $ac_cv_func__Block_copy = yes ; then
+    HAVE_BLOCKS=1
+  else
+    HAVE_BLOCKS=0
+    if test "$OBJC_RUNTIME_LIB" = "ng"; then
+      as_fn_error $? "The objc runtime library does not appear to have blocks support.  Try re-configuring gnustep-make with a CPPFLAGS variable containing a -L point to specify the directory containing the correct libobjc, or using the --with-objc-lib-flag=... option." "$LINENO" 5
+    fi
   fi
 fi
 
-
 # Don't revert any Objective-C flags as they are used in the next test
 
 GS_NONFRAGILE=0
 GS_MIXEDABI=0
 if test "$nonfragile" = "yes"; then
-  saved_CPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="$CPPFLAGS $OBJCFLAGS -fobjc-nonfragile-abi -x objective-c"
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for non-fragile-abi support" >&5
+  if test "$host_os" = windows -a "$OBJC_RUNTIME_LIB" = "ng"; then
+    # On Windows MSVC, AC_CHECK_FUNCS() will throw linker errors ("relocation
+    # against symbol in discarded section") as the file won't contain any actual
+    # ObjC code, so we just assume the function is there.
+    { $as_echo "$as_me:${as_lineno-$LINENO}: assuming non-fragile-abi support on Windows MSVC" >&5
+$as_echo "$as_me: assuming non-fragile-abi support on Windows MSVC" >&6;}
+    GS_NONFRAGILE=1
+  else
+    saved_CPPFLAGS="$CPPFLAGS"
+    CPPFLAGS="$CPPFLAGS $OBJCFLAGS -fobjc-nonfragile-abi -x objective-c"
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for non-fragile-abi support" >&5
 $as_echo_n "checking for non-fragile-abi support... " >&6; }
-  if test "$cross_compiling" = yes; then :
-  non_fragile="$cross_non_fragile"
-else
+    if test "$cross_compiling" = yes; then :
+      non_fragile="$cross_non_fragile"
+    else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include "$srcdir/config/config.non-fragile-ivars.m"
 _ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  non_fragile=yes
-else
-  non_fragile=no
-fi
+      if ac_fn_c_try_run "$LINENO"; then :
+        non_fragile=yes
+      else
+        non_fragile=no
+      fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
+    fi
 
-  CPPFLAGS="$saved_CPPFLAGS"
-  if test $non_fragile = yes; then
-    OBJCLIBS="$OBJLIBS -fobjc-nonfragile-abi"
-    GS_NONFRAGILE=1
-    if test "$OBJC_RUNTIME_LIB" != "ng"; then
-      # Check whether --enable-mixedabi was given.
-if test "${enable_mixedabi+set}" = set; then :
-  enableval=$enable_mixedabi;
-else
-  enable_mixedabi=yes
-fi
+    CPPFLAGS="$saved_CPPFLAGS"
+    if test $non_fragile = yes; then
+      OBJCLIBS="$OBJLIBS -fobjc-nonfragile-abi"
+      GS_NONFRAGILE=1
+      if test "$OBJC_RUNTIME_LIB" != "ng"; then
+        # Check whether --enable-mixedabi was given.
+        if test "${enable_mixedabi+set}" = set; then :
+          enableval=$enable_mixedabi;
+        else
+          enable_mixedabi=yes
+        fi
 
-      if test $enable_mixedabi = yes; then
-        GS_MIXEDABI=1
+        if test $enable_mixedabi = yes; then
+          GS_MIXEDABI=1
+        fi
       fi
     fi
-  fi
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $non_fragile" >&5
+    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $non_fragile" >&5
 $as_echo "$non_fragile" >&6; }
+  fi
 fi
 
 
 
-
 # Don't revert any Objective-C flags as they are used in the next test
 
 #--------------------------------------------------------------------
@@ -8037,21 +8068,30 @@ $as_echo "$have_set_unexpected" >&6; }
 # Don't revert any Objective-C flags as they are used in the next test
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for _objc_unexpected_exception in runtime" >&5
-$as_echo_n "checking for _objc_unexpected_exception in runtime... " >&6; }
-if test "$cross_compiling" = yes; then :
-  have_unexpected="$cross_have_unexpected"
+  $as_echo_n "checking for _objc_unexpected_exception in runtime... " >&6; }
+if test "$host_os" = windows -a "$OBJC_RUNTIME_LIB" = "ng"; then
+  # On Windows MSVC, AC_CHECK_FUNCS() will throw linker errors ("relocation
+  # against symbol in discarded section") as the file won't contain any actual
+  # ObjC code, so we just assume the function is there.
+  { $as_echo "$as_me:${as_lineno-$LINENO}: assuming _objc_unexpected_exception() on Windows MSVC" >&5
+$as_echo "$as_me: assuming _objc_unexpected_exception() on Windows MSVC" >&6;}
+  have_unexpected=yes
 else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+  if test "$cross_compiling" = yes; then :
+    have_unexpected="$cross_have_unexpected"
+  else
+    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include "$srcdir/config/config.unexpected.m"
 _ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  have_unexpected=yes
-else
-  have_unexpected=no
-fi
+    if ac_fn_c_try_run "$LINENO"; then :
+      have_unexpected=yes
+    else
+      have_unexpected=no
+    fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
+  fi
 fi
 
 if test $have_unexpected = yes; then
@@ -8099,24 +8139,32 @@ fi
 #--------------------------------------------------------------------
 # Function needed by @synchronize directive
 #--------------------------------------------------------------------
-for ac_func in objc_sync_enter
-do :
-  ac_fn_c_check_func "$LINENO" "objc_sync_enter" "ac_cv_func_objc_sync_enter"
-if test "x$ac_cv_func_objc_sync_enter" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
+HAVE_OBJC_SYNC_ENTER=no
+if test "$host_os" = windows -a "$OBJC_RUNTIME_LIB" = "ng"; then
+  # On Windows MSVC, AC_CHECK_FUNCS() will throw linker errors ("relocation
+  # against symbol in discarded section") as the file won't contain any actual
+  # ObjC code, so we just assume the function is there.
+  { $as_echo "$as_me:${as_lineno-$LINENO}: assuming objc_sync_enter() on Windows MSVC" >&5
+$as_echo "$as_me: assuming objc_sync_enter() on Windows MSVC" >&6;}
+  HAVE_OBJC_SYNC_ENTER=yes
+else
+  for ac_func in objc_sync_enter
+  do :
+    ac_fn_c_check_func "$LINENO" "objc_sync_enter" "ac_cv_func_objc_sync_enter"
+    if test "x$ac_cv_func_objc_sync_enter" = xyes; then :
+    cat >>confdefs.h <<_ACEOF
 #define HAVE_OBJC_SYNC_ENTER 1
 _ACEOF
 
-fi
-done
-
+    fi
+  done
 
-HAVE_OBJC_SYNC_ENTER=no
-if test $ac_cv_func_objc_sync_enter = yes ; then
-  HAVE_OBJC_SYNC_ENTER=yes
+  HAVE_OBJC_SYNC_ENTER=no
+  if test $ac_cv_func_objc_sync_enter = yes ; then
+    HAVE_OBJC_SYNC_ENTER=yes
+  fi
 fi
 
-
 # Don't revert any Objective-C flags as they are used in the next test
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for thread-safe +initialize in runtime" >&5
